src/runtime/virtcontainers/qemu.go

```diff
@@ -667,6 +672,18 @@ func (q *qemu) CreateVM(ctx context.Context, id string, network Network, hypervi
 		qemuConfig.Devices = q.arch.appendPCIeRootPortDevice(qemuConfig.Devices, hypervisorConfig.PCIeRootPort, memSize32bit, memSize64bit)
 	}
 
+	if q.config.Debug && !qemuConfig.Knobs.Daemonize {
+		qemuConfig.LogFunc =
+			func(qemuPid int, qemuStderr io.ReadCloser) {
```

Comment:
Would it be better to also pass in the sandbox id instead given that most of the logs have sandbox id?